In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# reading data as pandas dataframe
application = pd.read_csv('application_record.csv')
credit = pd.read_csv('credit_record.csv')

In [3]:
# lets look at our application data
application.head()

ID CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  CNT_CHILDREN  \
0  5008804           M            Y               Y             0   
1  5008805           M            Y               Y             0   
2  5008806           M            Y               Y             0   
3  5008808           F            N               Y             0   
4  5008809           F            N               Y             0   

   AMT_INCOME_TOTAL      NAME_INCOME_TYPE            NAME_EDUCATION_TYPE  \
0          427500.0               Working               Higher education   
1          427500.0               Working               Higher education   
2          112500.0               Working  Secondary / secondary special   
3          270000.0  Commercial associate  Secondary / secondary special   
4          270000.0  Commercial associate  Secondary / secondary special   

     NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  DAYS_BIRTH  DAYS_EMPLOYED  \
0        Civil marriage   Rented apartment      -12005          -4542   
1        Civil marriage   Rented apartment      -12005          -4542   
2               Married  House / apartment      -21474          -1134   
3  Single / not married  House / apartment      -19110          -3051   
4  Single / not married  House / apartment      -19110          -3051   

   FLAG_MOBIL  FLAG_WORK_PHONE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  \
0           1                1           0           0             NaN   
1           1                1           0           0             NaN   
2           1                0           0           0  Security staff   
3           1                0           1           1     Sales staff   
4           1                0           1           1     Sales staff   

   CNT_FAM_MEMBERS  
0              2.0  
1              2.0  
2              2.0  
3              1.0  
4              1.0

In [126]:
# lets look at our credit data
a = credit.groupby('ID').MONTHS_BALANCE.max() == 0
a[a == True]

ID
5001711    True
5001712    True
5001713    True
5001714    True
5001715    True
           ... 
5150481    True
5150483    True
5150484    True
5150485    True
5150487    True
Name: MONTHS_BALANCE, Length: 33856, dtype: bool

In [63]:
length = pd.DataFrame(credit.ID.value_counts()).rename(columns = {'ID': 'length'})

In [71]:
count = pd.DataFrame(credit.groupby(['ID', 'STATUS']).size().unstack().reset_index())

In [75]:
credit.groupby(['ID', 'STATUS']).size().unstack().reset_index()

STATUS       ID     0   1   2   3   4   5     C     X
0       5001711   3.0 NaN NaN NaN NaN NaN   NaN   1.0
1       5001712  10.0 NaN NaN NaN NaN NaN   9.0   NaN
2       5001713   NaN NaN NaN NaN NaN NaN   NaN  22.0
3       5001714   NaN NaN NaN NaN NaN NaN   NaN  15.0
4       5001715   NaN NaN NaN NaN NaN NaN   NaN  60.0
...         ...   ...  ..  ..  ..  ..  ..   ...   ...
45980   5150482  12.0 NaN NaN NaN NaN NaN   6.0   NaN
45981   5150483   NaN NaN NaN NaN NaN NaN   NaN  18.0
45982   5150484  12.0 NaN NaN NaN NaN NaN   1.0   NaN
45983   5150485   2.0 NaN NaN NaN NaN NaN   NaN   NaN
45984   5150487   NaN NaN NaN NaN NaN NaN  30.0   NaN

[45985 rows x 9 columns]

In [111]:
one_year_hist = pd.DataFrame(credit.groupby(['ID', 'MONTHS_BALANCE']).size().unstack().reset_index().iloc[:, 49:62])

In [112]:
one_year_hist

MONTHS_BALANCE  -12  -11  -10  -9   -8   -7   -6   -5   -4   -3   -2   -1   \
0               NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  1.0  1.0  1.0   
1               1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0   
2               1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0   
3               1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0   
4               1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0   
...             ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   
45980           1.0  1.0  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   
45981           1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0   
45982           1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0   
45983           NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  1.0   
45984           1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0   

MONTHS_BALANCE   0   
0               1.0  
1               1.0  
2               1.0  
3               1.0  
4               1.0  
...             ...  
45980           NaN  
45981           1.0  
45982           1.0  
45983           1.0  
45984           1.0  

[45985 rows x 13 columns]

In [109]:
one_year_hist.columns = ['12', '11', '10', '9', '8', '7', '6', '5', '4', '3', '2', 'last_month', 'this_month']

In [110]:
one_year_hist

12   11   10    9    8    7    6    5    4    3    2  last_month  \
0      NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  1.0  1.0         1.0   
1      1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0         1.0   
2      1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0         1.0   
3      1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0         1.0   
4      1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0         1.0   
...    ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...         ...   
45980  1.0  1.0  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN         NaN   
45981  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0         1.0   
45982  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0         1.0   
45983  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN         1.0   
45984  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0         1.0   

       this_month  
0             1.0  
1             1.0  
2             1.0  
3             1.0  
4             1.0  
...           ...  
45980         NaN  
45981         1.0  
45982         1.0  
45983         1.0  
45984         1.0  

[45985 rows x 13 columns]

In [149]:
credit[(credit.STATUS == '5') & (credit.MONTHS_BALANCE >= 0)].ID

70719      5004891
77253      5005205
108654     5009744
108761     5009749
127611     5010868
            ...   
981471     5142456
1003607    5145690
1014278    5146444
1029811    5149190
1040847    5149828
Name: ID, Length: 65, dtype: int64

In [159]:
credit.groupby(['ID', 'STATUS']).MONTHS_BALANCE.size().unstack().iloc[:,5:6]

STATUS    5
ID         
5001711 NaN
5001712 NaN
5001713 NaN
5001714 NaN
5001715 NaN
...      ..
5150482 NaN
5150483 NaN
5150484 NaN
5150485 NaN
5150487 NaN

[45985 rows x 1 columns]

In [5]:
# lets get some info about our data: like the data types, nulls, total counts
application.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 438557 entries, 0 to 438556
Data columns (total 18 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   ID                   438557 non-null  int64  
 1   CODE_GENDER          438557 non-null  object 
 2   FLAG_OWN_CAR         438557 non-null  object 
 3   FLAG_OWN_REALTY      438557 non-null  object 
 4   CNT_CHILDREN         438557 non-null  int64  
 5   AMT_INCOME_TOTAL     438557 non-null  float64
 6   NAME_INCOME_TYPE     438557 non-null  object 
 7   NAME_EDUCATION_TYPE  438557 non-null  object 
 8   NAME_FAMILY_STATUS   438557 non-null  object 
 9   NAME_HOUSING_TYPE    438557 non-null  object 
 10  DAYS_BIRTH           438557 non-null  int64  
 11  DAYS_EMPLOYED        438557 non-null  int64  
 12  FLAG_MOBIL           438557 non-null  int64  
 13  FLAG_WORK_PHONE      438557 non-null  int64  
 14  FLAG_PHONE           438557 non-null  int64  
 15  FLAG_EMAIL       

In [6]:
credit.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1048575 entries, 0 to 1048574
Data columns (total 3 columns):
 #   Column          Non-Null Count    Dtype 
---  ------          --------------    ----- 
 0   ID              1048575 non-null  int64 
 1   MONTHS_BALANCE  1048575 non-null  int64 
 2   STATUS          1048575 non-null  object
dtypes: int64(2), object(1)
memory usage: 24.0+ MB


In [7]:
# lets get value counts and the count of unique observations in each column
for col in application.columns:
    print(col)
    print(application[col].value_counts())
    print(f'the number of unique observation for the col is: {application[col].nunique()}')
    print()

ID
7603224    2
7742853    2
7836711    2
7091721    2
7022327    2
          ..
5372264    1
6412652    1
6410605    1
6367602    1
6291456    1
Name: ID, Length: 438510, dtype: int64
the number of unique observation for the col is: 438510

CODE_GENDER
F    294440
M    144117
Name: CODE_GENDER, dtype: int64
the number of unique observation for the col is: 2

FLAG_OWN_CAR
N    275459
Y    163098
Name: FLAG_OWN_CAR, dtype: int64
the number of unique observation for the col is: 2

FLAG_OWN_REALTY
Y    304074
N    134483
Name: FLAG_OWN_REALTY, dtype: int64
the number of unique observation for the col is: 2

CNT_CHILDREN
0     304071
1      88527
2      39884
3       5430
4        486
5        133
7          9
9          5
12         4
6          4
14         3
19         1
Name: CNT_CHILDREN, dtype: int64
the number of unique observation for the col is: 12

AMT_INCOME_TOTAL
135000.0    46879
157500.0    39768
180000.0    37912
112500.0    36979
225000.0    34130
            ...  
300600.0

In [8]:
# lets get value counts and the count of unique observations in each column
for col in credit.columns:
    print(col)
    print(credit[col].value_counts())
    print(f'the number of unique observation for the col is: {credit[col].nunique()}')
    print()

ID
5145767    61
5116236    61
5037153    61
5002523    61
5028973    61
           ..
5149044     1
5079047     1
5137258     1
5023604     1
5023029     1
Name: ID, Length: 45985, dtype: int64
the number of unique observation for the col is: 45985

MONTHS_BALANCE
-1     34436
-2     34209
 0     33856
-3     33854
-4     33365
       ...  
-56     2109
-57     1649
-58     1255
-59      820
-60      415
Name: MONTHS_BALANCE, Length: 61, dtype: int64
the number of unique observation for the col is: 61

STATUS
C    442031
0    383120
X    209230
1     11090
5      1693
2       868
3       320
4       223
Name: STATUS, dtype: int64
the number of unique observation for the col is: 8



In [9]:
# creating  list of unique observations for both dataframes
unique_application_id = list(application.ID.unique())
unique_credit_id = list(credit.ID.unique())

In [10]:
len(unique_credit_id)

45985

In [11]:
# checking how many of the unique ids in application dataframe match the unique ids in credit dataframe

# appication_and_credit = []
# just_application= []
# for val in unique_application_id:
#     if val in unique_credit_id:
#         appication_and_credit.append(val)
#     else:
#         just_application.append(val)

In [12]:
# IDs in both application and credit
# len(appication_and_credit)

In [13]:
# # IDs in just application
# len(just_application)

# Takeaways:
- there are 438510 unique  ids  in application dataframe
- there are 45985 unique ids in credit dataframe
- Out of 438510 unique ids in application dataframe, only 36457 of them have records of their credit activity


**Thought Process For Filtering the data:**
- There are 438510 unique ids in applicant data frame but there are only 45985 unique id in credit history dataframe
- When cross validated, only 36457 of the unique ids of the applicant dataframe have credit history. 
- So, lets create a dataframe of applicant whose id are in credit history data frame and filter the credit histroy dataframe to have just the history of these applicant
- Also, save a copy locally

In [14]:
application

ID CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  CNT_CHILDREN  \
0       5008804           M            Y               Y             0   
1       5008805           M            Y               Y             0   
2       5008806           M            Y               Y             0   
3       5008808           F            N               Y             0   
4       5008809           F            N               Y             0   
...         ...         ...          ...             ...           ...   
438552  6840104           M            N               Y             0   
438553  6840222           F            N               N             0   
438554  6841878           F            N               N             0   
438555  6842765           F            N               Y             0   
438556  6842885           F            N               Y             0   

        AMT_INCOME_TOTAL      NAME_INCOME_TYPE            NAME_EDUCATION_TYPE  \
0               427500.0               Working               Higher education   
1               427500.0               Working               Higher education   
2               112500.0               Working  Secondary / secondary special   
3               270000.0  Commercial associate  Secondary / secondary special   
4               270000.0  Commercial associate  Secondary / secondary special   
...                  ...                   ...                            ...   
438552          135000.0             Pensioner  Secondary / secondary special   
438553          103500.0               Working  Secondary / secondary special   
438554           54000.0  Commercial associate               Higher education   
438555           72000.0             Pensioner  Secondary / secondary special   
438556          121500.0               Working  Secondary / secondary special   

          NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  DAYS_BIRTH  DAYS_EMPLOYED  \
0             Civil marriage   Rented apartment      -12005          -4542   
1             Civil marriage   Rented apartment      -12005          -4542   
2                    Married  House / apartment      -21474          -1134   
3       Single / not married  House / apartment      -19110          -3051   
4       Single / not married  House / apartment      -19110          -3051   
...                      ...                ...         ...            ...   
438552             Separated  House / apartment      -22717         365243   
438553  Single / not married  House / apartment      -15939          -3007   
438554  Single / not married       With parents       -8169           -372   
438555               Married  House / apartment      -21673         365243   
438556               Married  House / apartment      -18858          -1201   

        FLAG_MOBIL  FLAG_WORK_PHONE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  \
0                1                1           0           0             NaN   
1                1                1           0           0             NaN   
2                1                0           0           0  Security staff   
3                1                0           1           1     Sales staff   
4                1                0           1           1     Sales staff   
...            ...              ...         ...         ...             ...   
438552           1                0           0           0             NaN   
438553           1                0           0           0        Laborers   
438554           1                1           0           0     Sales staff   
438555           1                0           0           0             NaN   
438556           1                0           1           0     Sales staff   

        CNT_FAM_MEMBERS  
0                   2.0  
1                   2.0  
2                   2.0  
3                   1.0  
4                   1.0  
...                 ...  
438552              1.0  
438553              1.0  
438554              1.0  
43855

In [15]:
# so lets get a dataframe that holds all of the observation that are unique in all of the features

non_dup_app = application[application.duplicated(subset =['CODE_GENDER', 'FLAG_OWN_CAR', 'FLAG_OWN_REALTY', 'CNT_CHILDREN',
       'AMT_INCOME_TOTAL', 'NAME_INCOME_TYPE', 'NAME_EDUCATION_TYPE',
       'NAME_FAMILY_STATUS', 'NAME_HOUSING_TYPE', 'DAYS_BIRTH',
       'DAYS_EMPLOYED', 'FLAG_MOBIL', 'FLAG_WORK_PHONE', 'FLAG_PHONE',
       'FLAG_EMAIL', 'OCCUPATION_TYPE', 'CNT_FAM_MEMBERS'] ) == False]

In [16]:
# dataframe of non dup observations
non_dup_app

ID CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  CNT_CHILDREN  \
0       5008804           M            Y               Y             0   
2       5008806           M            Y               Y             0   
3       5008808           F            N               Y             0   
7       5008812           F            N               Y             0   
10      5008815           M            Y               Y             0   
...         ...         ...          ...             ...           ...   
438541  6837707           M            N               Y             0   
438545  6839651           F            N               Y             3   
438547  6839917           F            N               Y             0   
438552  6840104           M            N               Y             0   
438553  6840222           F            N               N             0   

        AMT_INCOME_TOTAL      NAME_INCOME_TYPE            NAME_EDUCATION_TYPE  \
0               427500.0               Working               Higher education   
2               112500.0               Working  Secondary / secondary special   
3               270000.0  Commercial associate  Secondary / secondary special   
7               283500.0             Pensioner               Higher education   
10              270000.0               Working               Higher education   
...                  ...                   ...                            ...   
438541          202500.0               Working               Higher education   
438545           99000.0             Pensioner  Secondary / secondary special   
438547          180000.0             Pensioner               Higher education   
438552          135000.0             Pensioner  Secondary / secondary special   
438553          103500.0               Working  Secondary / secondary special   

          NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  DAYS_BIRTH  DAYS_EMPLOYED  \
0             Civil marriage   Rented apartment      -12005          -4542   
2                    Married  House / apartment      -21474          -1134   
3       Single / not married  House / apartment      -19110          -3051   
7                  Separated  House / apartment      -22464         365243   
10                   Married  House / apartment      -16872           -769   
...                      ...                ...         ...            ...   
438541        Civil marriage  House / apartment      -13510          -2309   
438545  Single / not married  House / apartment      -18832         365243   
438547               Married  House / apartment      -10966          -2704   
438552             Separated  House / apartment      -22717         365243   
438553  Single / not married  House / apartment      -15939          -3007   

        FLAG_MOBIL  FLAG_WORK_PHONE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  \
0                1                1           0           0             NaN   
2                1                0           0           0  Security staff   
3                1                0           1           1     Sales staff   
7                1                0           0           0             NaN   
10               1                1           1           1     Accountants   
...            ...              ...         ...         ...             ...   
438541           1                1           0           0        Laborers   
438545           1                0           0           0             NaN   
438547           1                0           0           0             NaN   
438552           1                0           0           0             NaN   
438553           1                0           0           0        Laborers   

        CNT_FAM_MEMBERS  
0                   2.0  
2                   2.0  
3                   1.0  
7                   1.0  
10                  2.0  
...                 ...  
438541              2.0  
438545              1.0  
438547              2.0  
43855

In [17]:
# filling null values for occupation with other
application = application.fillna('other')

In [19]:
# # creating a group of ids that share similar values for all of the other columns
# duplicated_id = {'groups':[]}
# list_ids = list(non_dup_app.ID)

# for val in list_ids:
#     df = application[application.ID == val]
#     df2 = application[(df[['CODE_GENDER', 'FLAG_OWN_CAR', 'FLAG_OWN_REALTY', 'CNT_CHILDREN',
#        'AMT_INCOME_TOTAL', 'NAME_INCOME_TYPE', 'NAME_EDUCATION_TYPE',
#        'NAME_FAMILY_STATUS', 'NAME_HOUSING_TYPE', 'DAYS_BIRTH',
#        'DAYS_EMPLOYED', 'FLAG_MOBIL', 'FLAG_WORK_PHONE', 'FLAG_PHONE',
#        'FLAG_EMAIL', 'OCCUPATION_TYPE', 'CNT_FAM_MEMBERS']].values == application[['CODE_GENDER', 'FLAG_OWN_CAR', 'FLAG_OWN_REALTY', 'CNT_CHILDREN',
#        'AMT_INCOME_TOTAL', 'NAME_INCOME_TYPE', 'NAME_EDUCATION_TYPE',
#        'NAME_FAMILY_STATUS', 'NAME_HOUSING_TYPE', 'DAYS_BIRTH',
#        'DAYS_EMPLOYED', 'FLAG_MOBIL', 'FLAG_WORK_PHONE', 'FLAG_PHONE',
#        'FLAG_EMAIL', 'OCCUPATION_TYPE', 'CNT_FAM_MEMBERS']].values).sum(axis = 1) == 17]
#     ids = list(df2.ID)
#     duplicated_id['groups'].append(ids)
    

In [ ]:
# Extracting the unique id from the above data frame and the credit dataframe

unique_application_id = list(non_dup_app.ID.unique())
unique_credit_id = list(credit.ID.unique())

In [ ]:
# here we are checking if the unique id in our application data frame is in the list of unique credit id list
# and creating a new list that holds IDs that are both in non duplicated dataframe and the credit id

# appication_and_credit = []
# just_application= []
# for val in unique_application_id:
#     if val in unique_credit_id:
#         appication_and_credit.append(val)
#     else:
#         just_application.append(val)

In [ ]:
# and now we are creating an application dataframe using the list of IDs that have both application and credit hist


# df = pd.DataFrame(columns = application.columns)
# for val in appication_and_credit:
#     df = df.append(application[application.ID == val])


In [ ]:
# saving the dataframe locally

# df.to_csv('app.csv')

In [ ]:
# and now we are creating an credit dataframe using the list of IDs that have both application and credit hist


# df = pd.DataFrame(columns = credit.columns)
# for val in appication_and_credit:
#     df = df.append(credit[credit.ID == val])

In [ ]:
# saving the data locally

# df.to_csv('cred.csv')

In [ ]:
from wrangle import get_application_data
from wrangle import get_reports_data
import preprocessing as prep

In [ ]:
apps = get_application_data('app.csv')

In [ ]:
apps.head()

In [ ]:
expanded, score, full_history = get_reports_data('cred.csv')

In [ ]:
expanded

In [ ]:
score

In [ ]:
full_history

In [ ]:
pd.read_csv('cred.csv')

**Now I want to create a list or maybe a dictionary to create a spearate group of the IDs from our initial application data frame that shared the same value for all of the other variables**

In [ ]:
duplicated_ids = list(application[application.duplicated(subset =['CODE_GENDER', 'FLAG_OWN_CAR', 'FLAG_OWN_REALTY', 'CNT_CHILDREN',
       'AMT_INCOME_TOTAL', 'NAME_INCOME_TYPE', 'NAME_EDUCATION_TYPE',
       'NAME_FAMILY_STATUS', 'NAME_HOUSING_TYPE', 'DAYS_BIRTH',
       'DAYS_EMPLOYED', 'FLAG_MOBIL', 'FLAG_WORK_PHONE', 'FLAG_PHONE',
       'FLAG_EMAIL', 'OCCUPATION_TYPE', 'CNT_FAM_MEMBERS'] )].ID)

In [ ]:
dup_df = application[application.duplicated(subset =['CODE_GENDER', 'FLAG_OWN_CAR', 'FLAG_OWN_REALTY', 'CNT_CHILDREN',
       'AMT_INCOME_TOTAL', 'NAME_INCOME_TYPE', 'NAME_EDUCATION_TYPE',
       'NAME_FAMILY_STATUS', 'NAME_HOUSING_TYPE', 'DAYS_BIRTH',
       'DAYS_EMPLOYED', 'FLAG_MOBIL', 'FLAG_WORK_PHONE', 'FLAG_PHONE',
       'FLAG_EMAIL', 'OCCUPATION_TYPE', 'CNT_FAM_MEMBERS'] )]

In [ ]:
dup_df

In [ ]:
for i in duplicated_ids:
    if application
    

In [ ]:
application[application.duplicated(subset = ['CODE_GENDER', 'FLAG_OWN_CAR', 'FLAG_OWN_REALTY', 'CNT_CHILDREN',
       'AMT_INCOME_TOTAL', 'NAME_INCOME_TYPE', 'NAME_EDUCATION_TYPE',
       'NAME_FAMILY_STATUS', 'NAME_HOUSING_TYPE', 'DAYS_BIRTH',
       'DAYS_EMPLOYED', 'FLAG_MOBIL', 'FLAG_WORK_PHONE', 'FLAG_PHONE',
       'FLAG_EMAIL', 'OCCUPATION_TYPE', 'CNT_FAM_MEMBERS'], keep = False)]

In [ ]:
a = application.set_index(keys = ['CODE_GENDER', 'FLAG_OWN_CAR', 'FLAG_OWN_REALTY', 'CNT_CHILDREN',
       'AMT_INCOME_TOTAL', 'NAME_INCOME_TYPE', 'NAME_EDUCATION_TYPE',
       'NAME_FAMILY_STATUS', 'NAME_HOUSING_TYPE', 'DAYS_BIRTH',
       'DAYS_EMPLOYED', 'FLAG_MOBIL', 'FLAG_WORK_PHONE', 'FLAG_PHONE',
       'FLAG_EMAIL', 'OCCUPATION_TYPE', 'CNT_FAM_MEMBERS'])

In [ ]:
a.index.duplicated()

In [ ]:
# checking anthony's wrangle function
import anthony_wrangle

In [ ]:
a, b , c = anthony_wrangle.get_reports_data('credit_record.csv')

In [ ]:
a.head()

In [ ]:
b.head()

In [ ]:
c.head()